### ISSUE TITLE
Compose fails to use registry credentials

### ISSUE THREAD

**Issue by birla** [containers/podman#22682]
### Bug description

Compose extension (which uses github.com/docker/compose) is not able to access registry credentials configured in podman.

Quick fix: pull all required images manually, using `podman pull privateregistry.com/image:tag` then run `podman compose up` and it works because the local image cache is seeded now.

### Operating system

Windows 11

### Installation Method

Installer from website/GitHub releases

### Version

1.9.0

### Steps to reproduce

1. Install Podman Desktop and dependencies.
2. Setup Compose extension (system-wide) using [docker compose](https://github.com/docker/compose/releases/tag/v2.26.1).
3. Login into a private container registry using either `podman login` or using the Podman desktop UI.
4. Verify that connection to the private registry is working by pulling an image using `podman pull` or the Podman desktop UI.
5. Try to spin up a docker compose file which relies on any image from the private container registry, not already available locally. `podman compose up`

### Relevant log output

```shell
Error response from daemon: {"message":"unable to retrieve auth token: invalid username/password: unauthorized: authentication required, visit https://aka.ms/acr/authorization for more information."}
Error: executing C:\Users\username\AppData\Local\Microsoft\WindowsApps\docker-compose.exe -f .\docker-compose.yml up -d: exit status 18
```

**Comment by gautaz** [comment]
Hello, I have the exact same issue.
I also had a `~/.docker/config.json` file with credentials from a previous docker install.
Removing this file did also not fix the issue.
So for now I am working around this the same way as @birla.

I got this error after a `podman system reset`, I have no idea if doing this might reproduce the issue...
I operated this reset because of inconsistencies in ` ~/.local/share/containers` subsequent to a crash and I had to hard reboot.

**Comment by birla** [comment]
Everyone on my team is facing the same issue, we used docker desktop until the company asked us to migrate to podman instead due to licensing issues.

**Comment by rrmistry** [comment]
Eventually got this working by copying `~/.config/containers/auth.json` to `~/.docker/config.json`.

Not ideal and would like a reasonable solution.

**Comment by birla** [comment]
> Eventually got this working by copying `~/.config/containers/auth.json` to `~/.docker/config.json`.
> 
> Not ideal and would like a reasonable solution.

I can confirm, this worked for me too.

**Comment by rhatdan** [comment]
No ideal way to solve this.

**Comment by jakubsemerak** [comment]
We are facing the same issue in our company, the workaround from @rrmistry seems to work, but it is not ideal.

**Comment by empire-penguin** [comment]
> Eventually got this working by copying `~/.config/containers/auth.json` to `~/.docker/config.json`.
> 
> Not ideal and would like a reasonable solution.

We faced a similar problem, and I can confirm this solution works. However we wanted this to apply systemwide and not have each user do this. Here is what worked for us.

```bash
cp ~/.config/containers/auth.json /etc/docker/config.json
export DOCKER_CONFIG=/etc/docker
```

We put the environment variable in `/etc/profile.d/`. This makes sure that all users have the environment variable set.

Then you can run `docker compose build <SERVICE>`.

**Comment by novotnyJiri** [comment]
Encountered the issue as well.  Kudos to @rrmistry for the workaround. Works.

**Comment by gautaz** [comment]
OK, there is no `~/.config/containers/auth.json` in my environment (NixOS).

Reviewing the `podman-login` manpage, the right path is `${XDG_RUNTIME_DIR}/containers/auth.json`.
So a `ln -s ${XDG_RUNTIME_DIR}/containers/auth.json ~/docker/config.json` did the trick for me.

Thanks @rrmistry.

**Comment by CarstenLeue** [comment]
Suggestion https://github.com/containers/podman/issues/22682#issuecomment-2216754294 is working for me, too, but I was hoping for a podman only solution

@rhatdan

**Comment by rhatdan** [comment]
With Podman login you can tell it to put the auth.json file in ~/.docker/config.json.

I guess when using `podman compose` we could check if the link exists and if not create it.  But not much else we can do.

**Comment by dmurvihill** [comment]
Please advise the correct file path to delete on Windows. Thx.

**Comment by jacob-kenin** [comment]
I am running Ubuntu 24.04 starting from a clean install with just the docker engine. Based off the findings in https://github.com/containers/podman/issues/22682#issuecomment-2367652052 I did the following:

```
mkdir ~/.docker
ln -s ${XDG_RUNTIME_DIR}/containers/auth.json ~/.docker/config.json
```

And now I can successfully pull containers from GitHub Container Registry (GHCR) after authenticating.

**Comment by mfontana-elem** [comment]
> With Podman login you can tell it to put the auth.json file in ~/.docker/config.json.
> 
> I guess when using `podman compose` we could check if the link exists and if not create it. But not much else we can do.

Thanks for the answer. Can you explain, for us who don't know, why a proper solution can't be implemented by `podman`?

Is `podman compose` just calling `docker compose` with some plumbing to make the latter run using `podman` as the container manager? This is the only scenario I can imagine right know for which a non-hacky solution would be impossible.

**Comment by prakharbirla-ng** [comment]
> Thanks for the answer. Can you explain, for us who don't know, why a proper solution can't be implemented by `podman`?
> 
> Is `podman compose` just calling `docker compose` with some plumbing to make the latter run using `podman` as the container manager? This is the only scenario I can imagine right know for which a non-hacky solution would be impossible.

Yes, your assumption is correct. Podman uses an external provider for compose commands wiz docker compose.

**Comment by mattnathan** [comment]
As this ticket is where you end up after searching for `podman compose authentication required` I thought I'd mention my specific setup which isn't listed in any of the above posts.

I'm running on MacOS, have never installed Docker or Docker Desktop. I installed Podman via the Podman Desktop installer and am using default settings which uses docker-compose as the compose tool.

I've run `podman login` which succeeded. My private registry is AWS ECR.

The [podman-login man page](https://docs.podman.io/en/v5.1.0/markdown/podman-login.1.html) mentions that on mac the auth.json file is located in `$HOME/.config/containers/auth.json`. I have no `~/.docker/` folder. `XDG_RUNTIME_DIR` is not set.

For me on macOS this made the issue go away (thanks to [previous posts](https://github.com/containers/podman/issues/22682#issuecomment-2367652052)):

```
mkdir ~/.docker/
ln -s ~/.config/containers/auth.json ~/.docker/config.json
```

Hopefully this kind of thing will make its way into the podman cli at some point.

**Comment by ljouon** [comment]
Thanks @mattnathan.  That worked for me as well.

**Comment by whyayala** [comment]
> OK, there is no `~/.config/containers/auth.json` in my environment (NixOS).
> 
> Reviewing the `podman-login` manpage, the right path is `${XDG_RUNTIME_DIR}/containers/auth.json`. So a `ln -s ${XDG_RUNTIME_DIR}/containers/auth.json ~/.docker/config.json` did the trick for me.
> 
> Thanks [@rrmistry](https://github.com/rrmistry).

I'm on macOs and this is close to the solution I arrived at (`ln -s ~/.config/containers/auth.json ~/.docker/config.json`) before coming here to create an issue and finding this thread. Symlinking is probably the least encumber-some solution if you just want to fix your machine and not have to worry about it again.

As others have stated, the reason this happens is because `podman login` stores credentials in one place and the compose extension in podman desktop is just a wrapper for `docker compose` which looks for credentials in another location.

This was very annoying to troubleshoot and probably turns people off to podman.

**Comment by MohammedEsafi** [comment]
> Eventually got this working by copying `~/.config/containers/auth.json` to `~/.docker/config.json`.
> 
> Not ideal and would like a reasonable solution.

This workaround fixed it for me. that problem was rather annoying

**Comment by jn-pt** [comment]
> > Eventually got this working by copying `~/.config/containers/auth.json` to `~/.docker/config.json`.
> > Not ideal and would like a reasonable solution.
> 
> This workaround fixed it for me. that problem was rather annoying

This also work for me on a fresh installation of the latest Podman release. Thanks!
